### PR TITLE
Fail on missing braces

### LINKED ISSUE #4518: Add error prone check to enforce use of braces
### Description
Teku coding standard requires use of braces for all if/while/for etc cases, even when the body is only a single line.  We should add a custom error prone check to enforce this.

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Enable the MissingBraces errorprone check and fix all violations
Enable the Finally (no returns/throws in finally blocks) and TypeParameterUnusedInFormals checks now that they work on Java 12+.

fixes #4518 even though I thought it wasn't possible...

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/attestation/ValidateableAttestation.java [hunk 1]
```diff
@@ -38,12 +38,12 @@
   private final Supplier<Bytes32> hashTreeRoot;
   private final AtomicBoolean gossiped = new AtomicBoolean(false);
   private final boolean producedLocally;
+  private final OptionalInt receivedSubnetId;
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/attestation/ValidateableAttestation.java:41]: ok to lose `volatile` here?

**Review by tbenr — APPROVED** [review]
LGTM with a doublecheck

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/attestation/ValidateableAttestation.java (hunk 1 above)
Comment (ajsutton) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/attestation/ValidateableAttestation.java:41]: Yep, a `final` field can't change so can't be `volatile`.

Comment (tbenr) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/attestation/ValidateableAttestation.java:41]: fair enough

**Pull request merged**